salt/utils/files.py

```diff
@@ -407,7 +407,10 @@ def flopen(*args, **kwargs):
     with fopen(*args, **kwargs) as f_handle:
         try:
             if is_fcntl_available(check_sunos=True):
-                fcntl.flock(f_handle.fileno(), fcntl.LOCK_SH)
+                lock_type = fcntl.LOCK_SH
+                if salt.utils.platform.is_aix() and ('a' in args[1] or 'w' in args[1]):
```

Comment:
I am not a huge fan of putting this lookup overhead into 100% of read calls in Salt. Can't we make this an argument instead?

Reply:
Yeah, we use this function everywhere, that would impact performance. It's better if we introduce a `lock_type` argument with a default value of `fcntl.LOCK_SH` and then rely on AIX-specific code to pass `lock_type=LOCK_EX` when opening files.

Reply:
referring to https://github.com/saltstack/zh/issues/1409, 
Will attempt to leverage something at import time so that I can use that, but the issue being that Exclusive locks is only required if file opened for write, and I would not want to hurt performance for reads.  But if worried about performance, why is that is_fcntl_available constantly being checked when only really applies if Solaris.  

Could always rewrite such that AIX don't bother with is_fcntl_available and resultant minimized performance hit. Let me know your thoughts